src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs

```diff
@@ -126,6 +128,23 @@ protected override string GenerateCommandLineCommands ()
 		{
 			var cmd = GetCommandLineBuilder ();
 
+			if (NamespaceTransforms?.Any () == true) {
+				// ex: obj/Debug/generated/msbuild-metadata.xml
+				var transform_file = Path.Combine (OutputDirectory, "..", "msbuild-metadata.xml");
+
+				var xml = new XDocument ();
```

Comment:
I'm not sure you need the `XDocument` at all?  `XElement.WriteTo()` also works…

Reply:
Updated.